src/blocks/custom.rs

```diff
@@ -116,20 +123,20 @@ async fn update_bar(
     if json {
         match serde_json::from_str::<Input>(stdout).error("Invalid JSON") {
             Ok(input) => {
-                widget.set_icon(&input.icon)?;
-                widget.state = input.state;
                 text_empty = input.text.is_empty();
-                if let Some(short) = input.short_text {
-                    widget.set_texts(input.text, short);
-                } else {
-                    widget.set_text(input.text);
-                }
+                let short_text = input.short_text.unwrap_or_default();
+                widget.set_values(map!{
+                    "icon" => Value::icon(api.get_icon(&input.icon)?),
+                    "text" => Value::text(input.text),
+                    "short_text" => Value::text(short_text)
```

Comment:
Maybe `[if let Some(t) = short_text] "short_text" => Value::text(t),` (just not set `short_text` instead of setting it to an empty string)?